cmd/kubeadm/app/cmd/phases/join/controlplanejoin.go

```diff
@@ -108,6 +111,14 @@ func newMarkControlPlaneSubphase() workflow.Phase {
 	}
 }
 
+func newWaitControlPlaneComponentsSubphase() workflow.Phase {
+	return workflow.Phase{
+		Name:   "wait-control-plane-components",
+		Hidden: true,
```

Comment:
a new hidden phases seems ok.
i guess on join we are not calling WaitForAPI or have a dedicated phase to wait for the apiserver to come up?

Reply:
> a new hidden phases seems ok. i guess on join we are not calling WaitForAPI or have a dedicated phase to wait for the apiserver to come up?

Do you mean we need to add legacy `WaitForAPI` code when this feature gate is disabled in this hidden phase

Reply:
no, i'm just trying to remember what happens on join.
i guess we are not even calling WaitForAPI on join, so adding the new hidden phase seems OK.
later if we decide we can make it non-hidden.